pkg/controller/migplan/registry.go

```diff
@@ -99,6 +101,47 @@ func (r ReconcileMigPlan) ensureMigRegistries(plan *migapi.MigPlan) error {
 	return err
 }
 
+func (r ReconcileMigPlan) ensureRegistryHealth(plan *migapi.MigPlan) error {
+	nEnsured := 0
+	clusters, err := r.planClusters(plan)
+	if err != nil{
+		return liberr.Wrap(err)
+	}
+
+	for _, cluster := range clusters{
+
+		if !cluster.Status.IsReady() {
+			continue
+		}
+
+		client, err := cluster.GetClient(r)
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+
+		registryStatusUnhealthy, err := isRegistryPodUnhealthy(plan, client)
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+
+		if !registryStatusUnhealthy{
+			nEnsured++
+		}
+	}
+
+	if nEnsured == 2 {
+		plan.Status.SetCondition(migapi.Condition{
+			Type:     RegistriesHealthy,
+			Status:   True,
+			Category: migapi.Required,
+			Message:  RegistriesHealthyMessage,
+		})
+	}
+
```

Comment:
you need to set the `RegistriesHealthy` to `False` here

Reply:
or remove the condition, depending on the logic

Reply:
Done